doc.go

```diff
@@ -61,8 +59,16 @@ func (d *Document) newField(field index.Field) {
 		}
 	}
 
-	d.fieldTokenFreqs = append(d.fieldTokenFreqs, af)
-	d.fieldLens = append(d.fieldLens, field.AnalyzedLength())
+	fieldIdx, exists := d.fieldIndexes[field.Name()]
+	if !exists {
+		d.fieldIndexes[field.Name()] = len(d.fieldTokenFreqs)
+		d.fieldNames = append(d.fieldNames, field.Name())
+		d.fieldTokenFreqs = append(d.fieldTokenFreqs, af)
+		d.fieldLens = append(d.fieldLens, field.AnalyzedLength())
+	} else {
+		d.fieldTokenFreqs[fieldIdx].MergeAll(field.Name(), af)
```

Comment:
This is the only part I'm not sure about.  The issue is that MergeAll does something special with the remoteField argument that really is only intended for building up the composite field data.  My concern is that this subtlely breaks something, although I cannot explain what...

Reply:
Ya fair point.

Let me create a simple version of merge token frequencies to here.